legal-api/src/legal_api/resources/business/business.py

```diff
@@ -41,3 +45,66 @@ def get(identifier):
             return jsonify({'message': f'{identifier} not found'}), HTTPStatus.NOT_FOUND
 
         return jsonify(business=business.json())
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def post():
+        """Create an incorporation filing, return the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    def put(identifier):
+        """Edit an incorporation filing. REturn the filing."""
+        return BusinessResource._save_incorporation_filing(request.get_json(), request, identifier)
+
+    @staticmethod
+    def _save_incorporation_filing(incorporation_body, client_request, business_id=None):
+        """Create or update an incorporation filing. """
+        # Check that there is a JSON filing
+        if not incorporation_body:
+            return  {'message': f'No filing json data in body of post for incorporation'}, \
+                HTTPStatus.BAD_REQUEST
+        # Validate Json
+
+        temp_corp_num = incorporation_body['filing']['incorporationApplication']['nameRequest']['nrNumber']
+        # temp_corp_num = business_id
+        # If this is an update to an incorporation filing, a temporary business identifier is passed in
+        if business_id:
+            business = Business.find_by_identifier(business_id)
+            if not business:
+                return  {'message': f'No incorporation filing exists for id {business_id}'}, \
+                HTTPStatus.BAD_REQUEST
+        else:
+            # Ensure there are no current businesses with the NR/random identifier
+            business = Business.find_by_identifier(temp_corp_num)
+
+            if business:
+                return  {'message': f'Incorporation filing for {business_id} already exists'}, \
+                HTTPStatus.BAD_REQUEST
+            # Create an empty business record, to be updated by the filer
+            business = Business()
+            business.identifier = temp_corp_num
+            business.save()
+
+        # Ensure the business identifier matches the NR in the filing
+        err = validate(business, incorporation_body)
+        if err:
+            return err
+
+        filing = Filing.get_filings_by_type(business.id, 'incorporationApplication')
+        # There can only be zero or one incorporation filings, if there are none, this is an
+        # initial request for incorporation. Create and insert a filing.
+        if not filing:
+            filing = Filing()
+            # filing.filing_type = 'incorporationApplication'
+            filing.business_id = business.id
+        elif len(filing) > 1:
+            return {'message': 'more than one incorporation filing found for corp'}, HTTPStatus.BAD_REQUEST
+        else:
+            filing = filing[0]
+        filing.filing_json = incorporation_body
+        filing.save()
```

Comment:
As data is getting  persisted to multiple tables shouldnt this happen in a transactional way? ie commit() should be called only after both are added to the session

Reply:
Good point, I will make the change